velox/functions/sparksql/tests/MakeTimestampTest.cpp

```diff
@@ -173,5 +173,54 @@ TEST_F(MakeTimestampTest, errors) {
       "make_timestamp requires session time zone to be set.");
 }
 
+TEST_F(MakeTimestampTest, invalidTimezone) {
+  const auto microsType = DECIMAL(16, 6);
+  const auto year = makeFlatVector<int32_t>({2021, 2021, 2021, 2021, 2021});
+  const auto month = makeFlatVector<int32_t>({7, 7, 7, 7, 7});
+  const auto day = makeFlatVector<int32_t>({11, 11, 11, 11, 11});
+  const auto hour = makeFlatVector<int32_t>({6, 6, 6, -6, 6});
+  const auto minute = makeFlatVector<int32_t>({30, 30, 30, 30, 30});
+  const auto micros = makeNullableFlatVector<int64_t>(
+      {45678000, 1e6, 6e7, 59999999, std::nullopt}, microsType);
+  auto data = makeRowVector({year, month, day, hour, minute, micros});
+
+  // Invalid time zone from query config.
+  setQueryTimeZone("Invalid");
```

Comment:
It would be nice to add some validation to config to reject "bad" settings early on. Would you consider taking this up as a follow up?

Reply:
We can validate if the timezone name is valid when accessing in QueryConfig, do you think it makes sense?
https://github.com/facebookincubator/velox/blob/ecf811833da61d951e60f927f745b78969f27213/velox/core/QueryConfig.h#L481-L483

Reply:
Ideally, we'd validate the values when they are being set. 

CC: @amitkdutta